python/test/backend/compiler_inspection.py

```diff
@@ -0,0 +1,24 @@
+import os
+from triton import knobs
+
+
+def inspect_stages(self, stages, options, language, capability):
+    if os.getenv('TRITON_MOCK_INSPECT_STAGES', '0') != '0':
+        source_code = "# This is generated from Triton add_stages_inspection_hook"
+        full_name = os.path.join(knobs.cache.dump_dir, "inspect_stages.py")
+        with open(full_name, "w") as file:
+            file.write(source_code)
```

Comment:
I don't understand what this tests, you are writing a file here and verifying that it exist later? Seems a convoluted way to check that this code is called?

Reply:
Yeah, I was trying to see if the hook was called from the test itself. I think I could just set a global variable instead.

Reply:
the file part seems odd though and unrelated to the rest of the flow, as long as this code is executed in the order you want then writing a file or anything else should be fine. 
Again I think we would benefit from testing explicitly the contract of the API hooks rather than some specific use case, this way it will make it clear what can be changed upstream without breaking the API

Reply:
it depends what we want to enforce, if it is just testing that this hook is called during compilation you could just use a nonlocal variable and store all the information you need in there

Reply:
I dropped this test and did the check using stdout.

I still have an env var in my test so that the "golden" version doesnt setup the hook.

Reply:
if you need examples of how to test hooks you can look at how we test the cache hooks right now. I have a strong preference for not using file or envvar unless strictly necessary

Reply:
I think this was overly complicated by nature of the test forking. I can follow the example of the other hooks in that case.

Reply:
> if you need examples of how to test hooks you can look at how we test the cache hooks right now. I have a strong preference for not using file or envvar unless strictly necessary

I've taken another pass at the tests and dropped all the subprocess forking and environment variables. I followed the `jit_cache_hook` test in python/test/unit/runtime/test_launch.py